### PR TITLE
[FranceTV] Back-port and update extractors

### LINKED ISSUE #25996: Cannot download from france.tv
Hi I can't download video from https://www.france.tv/france-2/avignon/963901-cour-d-honneur-de-jerome-bel-avignon-2013.html

### LINKED ISSUE #29956: Download from France.TV no longer possible
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
C:\Users\DELL\video>youtube-dl -v https://www.france.tv/france-3/plus-belle-la-vie/plus-belle-la-vie-saison-18/2739781-episode-du-vendredi-10-septembre-2021.html
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.france.tv/france-3/plus-belle-la-vie/plus-belle-la-vie-saison-18/2739781-episode-du-vendredi-10-septembre-2021.html']
[debug] Encodings: locale cp936, fs mbcs, out cp936, pref cp936
[debug] youtube-dl version 2021.03.14
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.18362
[debug] exe versions: ffmpeg N-89508-g1f1207145a, ffprobe N-89508-g1f1207145a
[debug] Proxy map: {'http': 'http://127.0.0.1:52528', 'socks': 'socks://127.0.0.1:52527', 'https': 'https://127.0.0.1:52528'}
[FranceTVSite] 2739781-episode-du-vendredi-10-septembre-2021: Downloading webpage
[FranceTV] b44f8a0a-e316-4735-9e1f-0fdf9839e2e2: Downloading video JSON
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp8z3w9mz6\build\youtube_dl\extractor\common.py", line 634, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp8z3w9mz6\build\youtube_dl\YoutubeDL.py", line 2279, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 580, in http_response
  File "C:\Python\Python34\lib\urllib\request.py", line 508, in error
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Python\Python34\lib\urllib\request.py", line 588, in http_error_default
```

## Description

WRITE DESCRIPTION HERE

### LINKED ISSUE #32713: la1ere: french overseas (9 regions) public and free
### CONTEXT

French overseas public channels are available without authentication from the portal : https://la1ere.francetvinfo.fr/

_Apparently, all formats/manifests are the same as those from france.tv_ (extractor francetv.py): 
indeed, a few videos are even shared by both sites.
For its developers, it should be straightforward to **derive a new extractor la1ere.py from francetv.py** (only two URL templates/pages embed videos) - i really hope my feeling is right, and would gladly help if i had some knowledge in video streams and python.

**The video offer is free, very large and covers documentaries, news, fictions of many cultures. It would be a nice gift for those interested in those, as internet speed may be slow overseas, preventing streaming :)**

9 regions are covered (below, <REGION> refers to one of)

- guadeloupe
- guyane
- martinique
- mayotte
- nouvellecaledonie
- polynesie
- reunion
- saintpierremiquelon
- wallisfutuna

Here is the mosaic of videos available for the chosen region:
https://la1ere.francetvinfo.fr/REGION/programme-video/
(as said before, a few of them are also available at: https://www.france.tv/la1ere/REGION/toutes-les-videos/)

### TEMPLATES OF EMBEDDING URLs

In the end, final pages embedding videos may have 2 paths only:

- https://la1ere.francetvinfo.fr/REGION/programme-video/diffusion/PAGE
- https://la1ere.francetvinfo.fr/REGION/programme-video/SHOW/diffusion/PAGE

The latter is used for series, news ...
The extractor may be flexible and even not consider/whitelist the region list, letting it be free (prophylaxis/evolutions), as the substring "/diffusion/" already filters templates.

### EXAMPLES:

Documentary:
https://la1ere.francetvinfo.fr/martinique/programme-video/diffusion/4774522-origine-kongo.html

News:
https://la1ere.francetvinfo.fr/guadeloupe/programme-video/la1ere_guadeloupe_le-13h-en-guadeloupe/diffusion/5643549-emission-du-lundi-29-janvier-2024.html

### PR DESCRIPTION
**Pull request by dirkf**
<details><summary>Boilerplate: yt-dlp+own code, bug-fix+improvement</summary>

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/), except for code from _yt-dlp_, for which this or the below has been asserted  
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

</details>

### Description of your *pull request* and other information

*  FranceTVIE: updated
*  FranceTVSiteIE: updated
*  FranceTVInfoIE: updated, add la1ere, use franceinfo.fr
*  FranceTVInfoSportIE: legacy redirect extractor
*  FranceTVJeunesseIE: obsolete, not `_WORKING`
*  GenerationWhatIE: obsolete, not `_WORKING`
*  CultureboxIE: legacy redirect extractor

Supersedes, closes #29996
Supersedes, closes #32884
Resolves #25996
Resolves #29956
Resolves #32713

### CODE REVIEW

File: youtube_dl/extractor/francetv.py [hunk 1]
```diff
@@ -191,66 +216,223 @@ def sign(manifest_url, manifest_id):
...
-            video_id = qs.get('idDiffusion', [None])[0]
-            catalog = qs.get('catalogue', [None])[0]
-            if not video_id:
-                raise ExtractorError('Invalid URL', expected=True)
+        url, smuggled_data = unsmuggle_url(url, {})
+        video_id = self._match_id(url)
+        hostname = smuggled_data.get('hostname') or 'www.france.tv'
 
-        return self._extract_video(video_id, catalog)
+        return self._extract_video(video_id, hostname=hostname)
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:324]:
```suggestion

class FranceTVEmbedIE(FranceTVBaseIE):
    _VALID_URL = r'''(?x)
        https?://embed\.francetv\.fr(?:/*\?.*?\b(?P<ue>ue)=|/)
        (?P<id>[\da-f]{32})(?:(?(ue)&|/?[?#]).*)?$
    '''
    _TESTS = [{
        'url': 'http://embed.francetv.fr/?ue=7fd581a2ccf59d2fc5719c5c13cf6961',
        'add_ie': [FranceTVIE.ie_key()],
        'info_dict': {
            'id': 'NI_983319',
            'ext': 'mp4',
            'title': 'Le Pen Reims',
            'upload_date': '20170505',
            'timestamp': 1493981780,
            'duration': 16,
        },
        'params': {
            'skip_download': True,
        },
        'skip': 'Gone',
    }, {
        'url': 'https://embed.francetv.fr/ecc238d6df70c0e9d76972fe1890a0e5',
        'add_ie': [FranceTVIE.ie_key()],
        'info_dict': {
            'id': '41495932-eaa9-11ef-a8a1-57a09c50f7ce',
            'ext': 'mp4',
            'title': 'Pouvoir d’achat : les Français de plus en plus endettés - Émission du jeudi 13 février 2025',
            'timestamp': 1739484973,
            'upload_date': '20250213',
            'duration': 111,
            'thumbnail': r're:https?://[\w.]+\.fr(?:/[\w-]+)+\.jpg$',
        },
        'params': {
            'format': 'best/bestvideo',
            'skip_download': 'm3u8',
        },
    }]

    def _real_extract(self, url):
        video_id = self._match_id(url)

        video = self._download_json(
            'https://api-embed.webservices.francetelevisions.fr/v2/key/%s' % (video_id,),
            video_id)

        return self._make_url_result(video['video_id'], video.get('url_source'))


```

File: youtube_dl/extractor/francetv.py [hunk 2]
```diff
@@ -191,66 +216,271 @@ def sign(manifest_url, manifest_id):
...
+            'upload_date': '20230424',
+            'duration': 3146,
+            'thumbnail': r're:https?:/(?:/[\w.-]+)+\.jpg$',
+        },
+        'expected_warnings': [
+            'hlsnative has detected features it does not support',
+            'Failed to download (?:MPD manifest|m3u8 information)',
+        ],
+        'skip': 'Geo-restricted to FR',
+    }, {
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:462]:
```suggestion
```

File: youtube_dl/extractor/francetv.py [hunk 3]
```diff
@@ -191,66 +216,253 @@ def sign(manifest_url, manifest_id):
...
-            catalog = qs.get('catalogue', [None])[0]
-            if not video_id:
-                raise ExtractorError('Invalid URL', expected=True)
+        return self._extract_video(video_id, hostname=hostname)
+
+
+class FranceTVEmbedIE(FranceTVBaseIE):
+    _VALID_URL = r'''(?x)
+        https?://embed\.francetv\.fr(?:/*\?.*?\b(?P<ue>ue)=|/)
+        (?P<id>[\da-f]{32})(?:(?(ue)&|/?[?#]).*)?$
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:328]:
```suggestion
        https?://embed\.francetv\.fr(?:/?\?(?:.*&)?(?P<ue>ue)=|/)
        (?P<id>[\da-f]{32})(?:(?(ue)&|/?[?#]).*)?$
```

File: youtube_dl/extractor/francetv.py [hunk 4]
```diff
@@ -191,66 +216,253 @@ def sign(manifest_url, manifest_id):
...
-            video_id = qs.get('idDiffusion', [None])[0]
-            catalog = qs.get('catalogue', [None])[0]
-            if not video_id:
-                raise ExtractorError('Invalid URL', expected=True)
+        return self._extract_video(video_id, hostname=hostname)
+
+
+class FranceTVEmbedIE(FranceTVBaseIE):
+    _VALID_URL = r'''(?x)
+        https?://embed\.francetv\.fr(?:/*\?.*?\b(?P<ue>ue)=|/)
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:327]:
```suggestion
        https?://embed\.francetv\.fr(?:/?\?(?:.*&)?(?P<ue>ue)=|/)
```

File: youtube_dl/extractor/francetv.py [hunk 5]
```diff
@@ -191,66 +216,253 @@ def sign(manifest_url, manifest_id):
...
+            'https://api-embed.webservices.francetelevisions.fr/v2/key/%s' % (video_id,),
+            video_id)
+
+        return self._make_url_result(video['video_id'], video.get('url_source'))
 
-class FranceTVSiteIE(FranceTVBaseInfoExtractor):
+
+class FranceTVSiteIE(FranceTVBaseIE):
+    IE_NAME = 'francetv:site'
     _VALID_URL = r'https?://(?:(?:www\.)?france\.tv|mobile\.france\.tv)/(?:[^/]+/)*(?P<id>[^/]+)\.html'
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:375]:
```suggestion
    _VALID_URL = r'https?://(?:(?:www|mobile)\.)?france\.tv)/(?:[^/]+/)*(?P<id>[^/]+)\.html'
```

File: youtube_dl/extractor/francetv.py [hunk 6]
```diff
@@ -1,70 +1,132 @@
...
-class FranceTVBaseInfoExtractor(InfoExtractor):
-    def _make_url_result(self, video_or_full_id, catalog=None):
-        full_id = 'francetv:%s' % video_or_full_id
-        if '@' not in video_or_full_id and catalog:
-            full_id += '@%s' % catalog
-        return self.url_result(
-            full_id, ie=FranceTVIE.ie_key(),
-            video_id=video_or_full_id.split('@')[0])
+try:
+    callable(format_field)
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:36]:
```suggestion
    if not callable(format_field):
        raise NameError
```

File: youtube_dl/extractor/francetv.py [hunk 7]
```diff
@@ -191,66 +217,253 @@ def sign(manifest_url, manifest_id):
...
+            video_id)
+
+        return self._make_url_result(video['video_id'], video.get('url_source'))
 
-class FranceTVSiteIE(FranceTVBaseInfoExtractor):
-    _VALID_URL = r'https?://(?:(?:www\.)?france\.tv|mobile\.france\.tv)/(?:[^/]+/)*(?P<id>[^/]+)\.html'
+
+class FranceTVSiteIE(FranceTVBaseIE):
+    IE_NAME = 'francetv:site'
+    _VALID_URL = r'https?://(?:(?:www|mobile)\.)?france\.tv)/(?:[^/]+/)*(?P<id>[^/]+)\.html'
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:376]:
```suggestion
    _VALID_URL = r'https?://(?:(?:www|mobile)\.)?france\.tv/(?:[^/]+/)*(?P<id>[^/]+)\.html'
```

File: youtube_dl/extractor/francetv.py [hunk 8]
```diff
@@ -191,66 +217,253 @@ def sign(manifest_url, manifest_id):
...
+                    'is_live': 'is_live',
+                }))
+            elif code:
+                if code == 2009:
+                    self.raise_geo_restricted(countries=self._GEO_COUNTRIES)
+                elif code in (2015, 2017):
+                    # 2015: L'accès à cette vidéo est impossible. (DRM-only)
+                    # 2017: Cette vidéo n'est pas disponible depuis le site web mobile (b/c DRM)
+                    drm_formats = True
+                    continue
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:271]:
```suggestion
                    continue
                elif code == 2007:
                    # 2007: Ce direct est terminé.
                    raise ExtractorError(
                        'Live broadcast has ended; video unavailable.',
                        video_id=video_id, expected=True)
```

File: youtube_dl/extractor/francetv.py [hunk 9]
```diff
@@ -191,66 +217,260 @@ def sign(manifest_url, manifest_id):
...
-                raise ExtractorError('Invalid URL', expected=True)
+        return self._extract_video(video_id, hostname=hostname)
+
+
+class FranceTVEmbedIE(FranceTVBaseIE):
+    _VALID_URL = r'''(?x)
+        https?://embed\.francetv\.fr(?:/?\?(?:.*&)?(?P<ue>ue)=|/)
+        # Say (?:...|) instead of (?:...)? when ... ends .* to avoid
+        # python/cpython#62847 (fixed from at least 3.5 and late 2.7)
+        (?P<id>[\da-f]{32})(?:(?(ue)&|/?[?#]).*|)$```
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:336]:
```suggestion
        # Say (?:|...) instead of (?:...)? when ... ends .* to avoid
        # python/cpython#62847 (fixed from at least 3.5 and late 2.7)
        (?P<id>[\da-f]{32})(?:|(?(ue)&|/?[?#]).*)$
```

File: youtube_dl/extractor/francetv.py [hunk 10]
```diff
@@ -191,66 +217,260 @@ def sign(manifest_url, manifest_id):
...
+                info = merge_dicts(info, traverse_obj(video, {
+                    'duration': 'duration',
+                    'is_live': 'is_live',
+                }))
+            elif code:
+                if code == 2009:
+                    self.raise_geo_restricted(countries=self._GEO_COUNTRIES)
+                elif code in (2015, 2017):
+                    # 2015: L'accès à cette vidéo est impossible. (DRM-only)
+                    # 2017: Cette vidéo n'est pas disponible depuis le site web mobile (b/c DRM)
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:269]:
```suggestion
                elif code in (2015, 2017, 2019):
                    # 2015: L'accès à cette vidéo est impossible. (DRM-only)
                    # 2017: Cette vidéo n'est pas disponible depuis le site web mobile (b/c DRM)
                    # 2019: L'accès à cette vidéo est incompatible avec votre configuration. ... (DRM-only?)
```

File: youtube_dl/extractor/francetv.py [hunk 11]
```diff
@@ -540,7 +866,7 @@ def _real_extract(self, url):
         entries = []
         for item in playlist['items']:
             identity = item.get('identity')
-            if identity and isinstance(identity, compat_str):
+            if identity and isinstance(identity, str):
                 entries.append(self._make_url_result(identity))
 
         return self.playlist_result(entries, playlist_id)
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:872]:
```suggestion
        return self.playlist_result(entries, playlist_id)


class LumniBaseIE(FranceTVBaseIE):
    _VIDEO_DIV_RE = r'(<div\s[^>]*(?<!-)\bdata-factoryid\s*=\s*(?!""|\'\')\S[^>]*>)'


class LumniIE(LumniBaseIE):
    _VALID_URL = r'https?://(?:www\.)?lumni\.fr/video/(?P<id>[\w-]+)'
    _TESTS = [{
        'url': 'https://www.lumni.fr/video/l-homme-et-son-environnement-dans-la-revolution-industrielle',
        'add_ie': [FranceTVIE.ie_key()],
        'md5': '84ddbef8fede632091f201eda38192d9',
        'info_dict': {
            'id': 'd2b9a4e5-a526-495b-866c-ab72737e3645',
            'ext': 'mp4',
            'title': "L'homme et son environnement dans la révolution industrielle - L'ère de l'homme",
            'timestamp': 1725372585,
            'upload_date': '20240903',
            'thumbnail': r're:https?:/(?:/[\w.-]+)+\.jpg$',
            'duration': 230,
        },
        'params': {
            'format': 'best/bestvideo',
            'skip_download': 'm3u8',
        },
    }]

    def _real_extract(self, url):
        display_id = self._match_id(url)
        webpage = self._download_webpage(url, display_id)
        video_id = extract_attributes(self._search_regex(
            self._VIDEO_DIV_RE, webpage, 'video id'))['data-factoryid']
        return self._make_url_result(video_id, url=url)


class LumniPlaylistIE(LumniBaseIE):
    _VALID_URL = r'''(?x)
                    https?://
                        (?:www\.)?lumni\.fr/
                        (?:dossier|programme|serie)/
                        (?P<id>[0-9a-z-]+)
                    '''
    _TESTS = [{
        'url': 'https://www.lumni.fr/dossier/les-fondamentaux-vocabulaire',
        'add_ie': [FranceTVIE.ie_key()],
        'info_dict': {
            'id': 'les-fondamentaux-vocabulaire',
            'title': 're:(?i)^Les Fondamentaux : Vocabulaire$',
        },
        'playlist_mincount': 9
    }, {
        'url': 'https://www.lumni.fr/programme/the-rich-morning-show',
        'only_matching': True
    }, {
        'url': 'https://www.lumni.fr/serie/la-maison-lumni-college',
        'only_matching': True
    }]

    def _real_extract(self, url):
        playlist_id = self._match_id(url)

        webpage = self._download_webpage(url, playlist_id)

        playlist_title = self._og_search_title(webpage)

        def entries():
            for m in re.finditer(self._VIDEO_DIV_RE, webpage):
                video_id = extract_attributes(m.group(1)).get('data-factoryid')
                if video_id:
                    yield self._make_url_result(video_id, url=url)

        return self.playlist_result(
            entries(), playlist_id, playlist_title)
```

File: youtube_dl/extractor/francetv.py [hunk 12]
```diff
@@ -470,7 +804,9 @@ def _real_extract(self, url):
         return self.url_result(youtube_id, ie='Youtube', video_id=youtube_id)
 
 
-class CultureboxIE(FranceTVBaseInfoExtractor):
+class CultureboxIE(FranceTVBaseIE):
+    # obsolete site: redirects to www.france.tv/spectacles-et-culture/
+    # specific URLs redirect under www.france.tv, eg /france-2/...
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:809]:
```suggestion
    # specific URLs redirect under www.france.tv, eg /france-2/...
    _WORKING = False
```

**Review by dcq01 — COMMENTED** [review]
Hi! I'm a grad student working on a research project about using large language models to automate code review. Based on your commit 7d2a0b425fed30603defec1f4cf7d0a7777f7f32 and the changes in youtube_dl/extractor/francetv.py, my tool generated this comment:
1. **Error Handling**: Implement robust error handling in the `_real_extract` method for cases where the API call fails or returns unexpected data.
2. **Potential Edge Cases**: Ensure that the logic that processes the matched groups handles cases where `ue` is not provided correctly.
3. **Validation of Extracted Data**: Ensure tests validate the extracted data against expected values, including `id`, `ext`, `title`, `upload_date`, `timestamp`, and `duration`.
4. **Sensitive Data Exposure**: Ensure that sensitive data, such as `video_id`, is not logged or exposed in error messages to prevent information disclosure vulnerabilities.
5. **Testing the Regex**: Add unit tests that cover various cases of URLs that should match and those that should not.
6. **Test Coverage for Regex Changes**: Ensure that the new regex correctly matches all intended URL formats and add new test cases for edge cases.
7. **Regular Expression Optimization**: Consider pre-compiling the regex using `re.compile()` to avoid recompilation on each call.
8. **Network Calls**: Implement caching mechanisms for the `_download_json` method to store previously fetched results, reducing the need for repeated network requests.

As part of my research, I'm trying to understand how useful these comments are in real-world development. If you have a moment, I'd be super grateful if you could quickly reply to these two yes/no questions:
1. Does this comment provide suggestions from a dimension you hadn’t considered?
2. 2. Do you find this comment helpful?

Thanks a lot for your time and feedback! And sorry again if this message is a bother.

**Comment by dirkf** [comment]
> ... I'd be super grateful if you could quickly reply to these two yes/no questions:
> 
>     1. Does this comment provide suggestions from a dimension you hadn’t considered?

No.

>     2. Do you find this comment helpful?

The suggestions cover known best practices and/or optimisations that are or would be implemented if appropriate and worthwhile. Item 4 is just wrong to identify `video_id` as sensitive.

File: youtube_dl/extractor/francetv.py [hunk 13]
```diff
@@ -394,23 +687,64 @@ def _real_extract(self, url):
...
+                'series': 'data-emission-title',
+                'url': ('data-url', T(url_or_none)),
+            }), {
+                'thumbnail': self._og_search_thumbnail(webpage, default=None),
+            })
+            if result.get('url'):
+                if not result.get('title'):
+                    result['title'] = self._html_search_regex(
+                        r'''Retrouvez l'intégralité du\s+(.+)\s*:''',
+                        webpage, 'Alt title').replace('"', '')
```
Comment (dirkf) [inline youtube_dl/extractor/francetv.py:720]: Support video ID in SvelteKit JS - pt 1:
```suggestion
        svelte_js_content = self._search_json(
            r'\bcontent\s*:', webpage, 'svelte content', display_id,
            transform_source=js_to_json, default={})
        result['id'] = traverse_obj(svelte_js_content, ('diffusion', 'videoId', T(str)))

        if not result.get('id'):
            player_button = extract_attributes(self._search_regex(
                r'(<button\s[^>]*(?<!-)\bdata-url\s*=\s*["\'][^>]{2,}>)',
                webpage, 'fi player button', default=''))
            if player_button.get('data-url'):
                result = merge_dicts(traverse_obj(player_button, {
                    'id': 'data-expression-uuid',
                    'timestamp': ('data-start-time', T(int_or_none)),
                    'duration': T(lambda x: int(x['data-end-time']) - int(x['data-start-time'])),
                    'title': 'data-extract-title',
                    'description': 'data-diffusion-title',
                    'series': 'data-emission-title',
                    'url': ('data-url', T(url_or_none)),
                }), {
                    'thumbnail': self._og_search_thumbnail(webpage, default=None),
                })
                if result.get('url'):
                    if not result.get('title'):
                        result['title'] = self._html_search_regex(
                            r'''Retrouvez l'intégralité du\s+(.+)\s*:''',
                            webpage, 'Alt title').replace('"', '')
```
